Template/discoveryService.json

```
@@ -0,0 +1,106 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "webAppName": {
+            "type": "string",
+            "minLength": 2
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]"
+        },
+        "sku": {
+            "type": "string",
+            "defaultValue": "B1",
+            "metadata": {
+                "description": "The SKU of App Service Plan."
+            }
+        },
+        "iotHubHostName": {
+            "type": "string"
+        },
+        "discoveryZipUrl": {
+            "type": "string"
+        },
+        "iotHubName": {
+            "type": "string"
+        },
+        "appInsightsInstrumentationKey": {
+            "type": "string"
+        },
+        "roleNameGuid": {
+            "type": "string",
+            "defaultValue": "[newGuid()]",
```

```suggestion
            "defaultValue": "[guid(resourceGroup().id, "twincontributor")]",
```

Someone might decide to reuse this template and if a new guid is generated, a "Conflict: RoleAssignmentExists" will be thrown.
Please consider usage of [guid function](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-string#guid) for scenarios like this one.

---

Nice catch, definitely better that way!